sheetshuttle/util.py

```diff
@@ -26,6 +26,17 @@ def gh_token_exists() -> bool:
     return False
 
 
+def extract_sheet_id(url: str) -> str:
+    """Extract a sheet ID from a Google Sheets URL."""
+    url_segments = url.split('/')
+    collect = False
+    for segment in url_segments:
+        if collect:
+            return segment
+        if segment == "d":
+            collect = True
```

Comment:
Another note on this function, we should make it so it raises an error if it cannot find an id in the expected location. The error message should be clear and tell the user that we can't find the Sheet id from the URL they gave us.
With that in mind, I think it might be worth reconsidering doing this with Regex since there are many cases to handle,  but it's up to you

Reply:
Relatively simplistic error handling has been added to `extract_sheet_id`--if you're expecting something a little more robust let me know.